src/utils/Hooks/Calculations/useAppropriateChartData.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useRef, useState } from 'react';
+import { differenceInMonths, differenceInWeeks } from 'date-fns';
+import { CategoryCodes, ChartData, MeasurementTypeCodesLabel, TimeUnitCodes } from '../../../types/chartDataTypes';
+
+export const useAppropriateChartData = (
+    chartDataForGender: ChartData,
+    dateOfBirth: Date,
+    defaultIndicator: string,
+    gender: string,
+    setDefaultIndicatorError: (value: boolean) => void,
+) => {
+    const [selectedCategory, setSelectedCategory] = useState<keyof typeof CategoryCodes>();
+    const [selectedDataset, setSelectedDataset] = useState<string>();
+
+    const selectDatasetForCategoryRef = useRef<(category: keyof typeof CategoryCodes) => void>();
+    selectDatasetForCategoryRef.current = (category: keyof typeof CategoryCodes) => {
+        const { datasets } = chartDataForGender[category];
+        const childAgeInWeeks = differenceInWeeks(new Date(), dateOfBirth);
```

Comment:
useMemo on big calculations that take up little storage, also for the childAgeInMonths

Reply:
This function is also used in the tooltip, can you extract this to GrowthChart.tsx and useMemo them here, then we will just have to do this calculation once for each child, if you think it is appropriate